apps/agent/contracts/SignatureValidator.sol

```diff
@@ -91,13 +91,17 @@ library SignatureValidator {
 
         output = new bytes(inputLength - 1);
 
+        if (output.length == 0) {
```

Comment:
> moving the output allocation within the if as well

Feel a little bit uncomfortable about this, since unallocated `bytes memory` variables start off assigned to the `0x60` memory slot (the [zero slot](https://solidity.readthedocs.io/en/v0.4.24/miscellaneous.html#layout-in-memory) that should never be written to). Assuming the consumer of this function uses the `bytes` variable properly (e.g. checking length, which should be done automatically...), everything should be fine, but passing around unallocated memory variables around still feels sketchy.

Had to double check that solidity would actually move the free memory pointer up if we allocated an empty array; would've been really bad otherwise.

I do agree we could put the `memcpy` inside the `if ()` if we wanted to save a bit of gas to avoid the manual `return`.

Reply:
Thanks for the comments, I think in this case I prefer having an explicit return of an empty array rather than an implicit return of an uninitialized array.